plugins/modules/s3_object.py

```diff
@@ -97,6 +97,13 @@
       - Keyname of the object inside the bucket.
       - Can be used to create "virtual directories", see examples.
     type: str
+  sig_v4:
+    description:
+      - Forces the Boto SDK to use Signature Version 4.
+      - Only applies to get modes, I(mode=get), I(mode=getstr), I(mode=geturl).
+    default: false
```

Comment:
What prevent us from turning this `True` by default?

Reply:
@goneri I have no issues with that - I was worried about it being a potential breaking change but on second thought it's just a URL. I can push this change right away.

Reply:
It more likes a question at this stage :-). I'm not sure if this can break people's playbooks. If we switch this `True` by default we will need to adjust the changelog fragment to mention the new behaviour.

@jillr, @tremble, what do you think?

Reply:
I think v4 has been set as the default by now but this will require research to confirm, and if that is true in all regions.  We also need to understand if there are any cases where this could break requests for older buckets or objects, and ensure we do the right thing for non-AWS S3-like APIs (such as Ceph RGW).

Reply:
For the AWS-like APIs there's code in there that bypasses the v4 signatures.  Definitely needs a changelog entry though if it's being enabled by default.

If AWS supports it on all calls, I'd be tempted to enable it by default everywhere rather than doing a per-mode test

Reply:
Sigv4 should be the default signing algorithm. See [Amazon's sigv2 deprecation plan](https://aws.amazon.com/blogs/aws/amazon-s3-update-sigv2-deprecation-period-extended-modified/)

Reply:
All, upon further research it appears that this is actually the default in recent versions of Boto.

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html

For the Config object, `signature_version='s3v4'` is the default if the param isn't passed, specifying `signature_version='s3'` would force SigV2.

This PR may be completely unnecessary.

Reply:
Ok! Had a chat with my team and here's what we came up with:

* If the region supports v2, Boto will default to it: https://github.com/boto/botocore/blob/73ce827de52ec52fd2f4e03dc2cdc105627edbfd/botocore/client.py#L425

Considering that the default behavior is to use V2 if the region has it enabled, otherwise use V4 - my opinion would be to allow the Ansible user to override this behavior in the event that they are deploying an a region with V2 enabled but would rather use V4. 

Example: AWS US Govcloud regions appear to support V2 and V4 so in this case, Boto would use V2.

I would set the default to `False` and rename the param to `force_sig_v4` and update the description to indicate that this only needs to be used if operating in a region where Boto's automatic detection would default to V2.

Reply:
It's also worth noting that in aws-us-gov, we want to force v4 because v2 is not FIPS compliant (as it uses SHA1).